examples/eng_wasm_contracts/voting_demo/src/lib.rs

```diff
@@ -0,0 +1,89 @@
+#![no_std]
+#![feature(proc_macro_gen)]
+#![feature(proc_macro_non_items)]
+
+
+extern crate eng_wasm;
+extern crate eng_wasm_derive;
+extern crate rustc_hex;
+#[macro_use]
+extern crate serde_derive;
+extern crate serde;
+
+use eng_wasm::*;
+use eng_wasm_derive::pub_interface;
+use eng_wasm_derive::eth_contract;
+use eng_wasm::String;
+use std::collections::HashMap;
+use std::string::ToString;
+
+#[eth_contract("VotingETH.json")]
+struct EthContract;
+
+// State key name "polls" holding a vector of Poll structs
+static POLLS: &str = "polls";
+// State key name "voting_eth_addr" holding eth address of VotingETH contract
+static VOTING_ETH_ADDR: &str = "voting_eth_addr";
+
+// Public-facing secret contract function declarations
+#[pub_interface]
+pub trait ContractInterface{
+    fn construct(voting_eth_addr: H160);
+    fn cast_vote(poll_id: U256, voter: H256, vote: U256);
+    fn tally_poll(poll_id: U256);
+}
+
+pub struct Contract;
+
+// Private functions accessible only by the secret contract
+impl Contract {
+    // Read secret contract state to obtain vector of Millionaires (or new vector if uninitialized)
+    fn get_polls() -> HashMap<u64, HashMap<String, U256>> {
+        read_state!(POLLS).unwrap_or_default()
+    }
+
+    fn get_voting_eth_addr() -> String {
+        read_state!(VOTING_ETH_ADDR).unwrap_or_default()
+    }
+}
+
+impl ContractInterface for Contract {
+    #[no_mangle]
+    fn construct(voting_eth_addr: H160) {
+        let voting_eth_addr_str: String = eformat!("{:?}", voting_eth_addr);
+        write_state!(VOTING_ETH_ADDR => voting_eth_addr_str);
+    }
+
+    #[no_mangle]
+    fn cast_vote(poll_id: U256, voter: H256, vote: U256) {
+        let mut polls = Self::get_polls();
+        eprint!("POLLS = {:?}", polls);
+        {
```

Comment:
what are these brackets for?

Reply:
I needed to use that to create a scope for the variables, otherwise was having some borrow issues